crates/exex/exex/src/backfill.rs

```diff
@@ -0,0 +1,126 @@
+use crate::ExExNotification;
+use reth_evm::execute::{BatchExecutor, BlockExecutorProvider};
+use reth_node_api::FullNodeComponents;
+use reth_primitives::BlockNumber;
+use reth_provider::{
+    BlockReader, Chain, DatabaseProviderFactory, HeaderProvider, HistoricalStateProviderRef,
+    ProviderError, StaticFileProviderFactory, TransactionVariant,
+};
+use reth_prune_types::PruneModes;
+use reth_revm::database::StateProviderDatabase;
+use reth_stages_types::ExecutionStageThresholds;
+use reth_tracing::tracing::{debug, trace};
+use std::{
+    ops::RangeInclusive,
+    sync::Arc,
+    time::{Duration, Instant},
+};
+use tokio::sync::mpsc::Sender;
+
+#[derive(Debug)]
+pub struct BackfillExecutor<Node: FullNodeComponents> {
+    components: Node,
+    notifications: Sender<ExExNotification>,
+    thresholds: ExecutionStageThresholds,
+}
+
+impl<Node: FullNodeComponents> BackfillExecutor<Node> {
+    pub fn new(components: Node, notifications: Sender<ExExNotification>) -> Self {
+        Self { components, notifications, thresholds: ExecutionStageThresholds::default() }
+    }
+}
+
+impl<Node: FullNodeComponents> BackfillExecutor<Node> {
+    pub async fn backfill(&self, range: RangeInclusive<BlockNumber>) -> eyre::Result<()> {
+        let start_block = *range.start();
```

Comment:
how do we expect this will be used?

this needs to be actively called but the notifications are emitted through the channel